pkg/util/docker/network.go

```diff
@@ -165,3 +173,57 @@ func DefaultGateway() (net.IP, error) {
 	}
 	return ip, nil
 }
+
+// GetAgentContainerNetworkMode provides the network mode of the Agent container
+// To get this info in an optimal way, consider calling util.GetAgentNetworkMode
+// instead to benefit from the cache
+func GetAgentContainerNetworkMode() (string, error) {
+	du, err := GetDockerUtil()
+	if err != nil {
+		return "", err
+	}
+	agentContainer, err := du.InspectSelf()
+	if err != nil {
+		return "", err
+	}
+	mode, err := parseContainerNetworkMode(agentContainer.HostConfig)
+	if err != nil {
+		return mode, err
+	}
+
+	// Try to discover awsvpc mode
+	if strings.HasPrefix(mode, containerModePrefix) {
+		// Inspect the attached container
+		co, err := du.Inspect(mode[len(containerModePrefix):], false)
+		if err != nil {
+			return "", fmt.Errorf("cannot inspect attached container %s: %v", mode, err)
+		}
+		// In awsvpc mode, the attached container is an amazon ecs pause container
+		if co.Config != nil && strings.HasPrefix(co.Config.Image, ecsPauseContainerImage) {
+			mode = containers.AwsvpcNetworkMode
+		}
+	}
+	return mode, nil
+}
+
+// parseContainerNetworkMode returns the network mode of a container
+func parseContainerNetworkMode(hostConfig *container.HostConfig) (string, error) {
+	if hostConfig == nil {
+		return "", errors.New("the HostConfig field is nil")
+	}
+	mode := string(hostConfig.NetworkMode)
+	switch mode {
+	case containers.DefaultNetworkMode:
+		return containers.DefaultNetworkMode, nil
+	case containers.HostNetworkMode:
+		return containers.HostNetworkMode, nil
+	case containers.BridgeNetworkMode:
+		return containers.BridgeNetworkMode, nil
+	case containers.NoneNetworkMode:
```

Comment:
these are the known docker network modes AFAIK, @hkaj wdyt?

Reply:
They will show up with the docker network name given to them, which gives an infinity of cases 😅do we want to support that?
In other words, if you run the agent with `--network <some_custom_network_name>` the `NetworkMode` show up with the <some_custom_network_name> value in the API

Reply:
My vote would be to return a new value  `containers.NetworkNetworkModeCustom` and let consumers discard that value if they need to, just so we don't have a chance of panicking. I don't have a sense of how often this will happen though.

Reply:
That's a good point, we should add a default.
It should do one of the following:

* return a unique value like @leeavital suggested, except i'd call it unknown instead of custom - because if users run a network driver that comes from a vendor they won't see it as custom. What matters in our logs / code is that we don't support it (yet), so `unknown` feels more accurate to me
* return an error if we can't handle that unknown mode nicely later in the code

Reply:
updated the PR accordingly 👍

Reply:
but I still have one question, when the Agent is attached to container network (`container:<id>` network mode) but not `awsvpc`, currently the returns the `container:<id>` string with `nil` error so consumers could rely on the attached container id (to inspect it or whatever). does this makes sense or would it be better to return `UnknownNetworkMode` with error if `container:<id>` doesn't corresponds to a awsvpc pause container ?